cloud/previews/previews.go

```diff
@@ -176,3 +210,55 @@ func (o CreatePreviewOpts) validate() error {
 
 	return errs.AsError()
 }
+
+func (o UpdateStackPreviewOpts) validate() error {
+	errs := errors.L()
+	if o.StackPreviewID == "" {
+		errs.Append(errors.E("stack preview id is empty"))
+	}
+
+	if string(o.OrgUUID) == "" {
+		errs.Append(errors.E("org uuid is empty"))
+	}
+
+	if o.Status == "" {
+		errs.Append(errors.E("status is empty"))
+	}
+
+	if o.ChangesetDetails != nil {
+		if o.ChangesetDetails.Provisioner == "" {
+			errs.Append(errors.E("provisioner is empty"))
+		}
+		if o.ChangesetDetails.ChangesetASCII == "" {
+			errs.Append(errors.E("changeset ascii is empty"))
+		}
+		if o.ChangesetDetails.ChangesetJSON == "" {
+			errs.Append(errors.E("changeset json is empty"))
+		}
+	}
+
+	return errs.AsError()
+}
+
+// DerivePreviewStatus derives the preview status from the exit code, run error
+// and whether the run is a terraform plan
+func DerivePreviewStatus(exitCode int, isTerraformPlan bool) StackPreviewStatus {
+	if isTerraformPlan {
+		switch exitCode {
+		case 2:
+			return PreviewStatusChanged
+		case 0:
+			return PreviewStatusUnchanged
+		}
+	}
+
+	switch exitCode {
+	case -1:
+		return PreviewStatusCanceled
+	case 0:
+		return PreviewStatusUnchanged
+	default:
+		return PreviewStatusFailed
+	}
+
+}
```

Comment:
This needs to be adjusted based on the discussion we had earlier.

Reply:
done